src/tests.rs

```diff
@@ -51,6 +51,22 @@ However, it does nothing.",
                 .symlink("/usr/bin/awesome"),
         )?
         .pre_install_script("echo preinst")
+        .scriptlet(
```

Comment:
I'm not sure we need two entirely different APIs to add a scriptlet.  The nice thing about an API like

`.pre_install_script(script: Into<Scriptlet>)`

 is that you could in theory do

`.pre_install_script("echo preinst")` if you have an `impl From<IntoString> for Scriptlet`

while still supporting more advanced things like

`.pre_install_script(
    Scriptlet::new("echo preinst")
        .flags(ScriptletFlags::EXPAND)
        .prog(vec!["/bin/blah/bash", "-c"])
)`

with the very same API.

Additionally it wouldn't seem like this meshes very well with the other functions?  For example, we don't do 

```
.dependency(Provides {...} )
.dependency(Requires {...} )
.dependency(Requires {...} )
.dependency(Requires {...} )
.dependency(Requires {...} )
```

and it seems like it would be less clear if we did use that kind of approach

Reply:
@dralley Thanks for taking the time to write this all out. It was honestly just a style choice I didn't have too strong of a reasoning behind it. I actually like your approach more.